caffe2/python/examples/resnet50_trainer.py

```diff
@@ -338,12 +365,14 @@ def test_input_fn(model):
                 test_reader,
                 batch_size=batch_per_device,
                 img_size=args.image_size,
+                dtype=args.dtype,
             )
 
         data_parallel_model.Parallelize(
             test_model,
             input_builder_fun=test_input_fn,
             forward_pass_builder_fun=create_resnet50_model_ops,
+            post_sync_builder_fun=add_post_sync_ops,
```

Comment:
It won't cause an error because it only adds nodes in the graph if there are fp16 and fp32 copies of weights.

When we generate the fp16 and fp32 weights, we then broadcast the _fp16_ copy to all GPUs, but then the fp16 and fp32 copies on a given GPU are out-of-sync. This adds some cast nodes to ensure that all fp16 and fp32 copies on all GPUs are the same.